src/pages/Organization/OrganizationView.tsx

```diff
@@ -28,20 +29,47 @@ interface Props {
 export default function OrganizationView({ id, navOrganizationId }: Props) {
   const { t } = useTranslation();
 
-  const [page, setPage] = useState(1);
-  const [searchQuery, setSearchQuery] = useState("");
-  const limit = 12; // 3x4 grid
+  const { qParams, updateQuery, Pagination, resultsPerPage } = useFilters({
+    limit: 12, // 3x4 grid
+    cacheBlacklist: ["name"],
+  });
+
+  const searchOptions = [
+    {
+      key: "name",
+      type: "text" as const,
+      placeholder: t("search_by_name"),
+      value: qParams.name || "",
+    },
+  ];
+
+  const handleSearch = useCallback(
+    (key: string, value: string) => {
+      const searchParams = {
+        name: key === "name" ? value : "",
+      };
+      updateQuery(searchParams);
+    },
+    [updateQuery],
+  );
+
+  const handleFieldChange = useCallback(() => {
+    updateQuery({
+      name: undefined,
+    });
+  }, [updateQuery]);
```

Comment:
checkout `removeFilters` from `useFilters`. also do we need this in the first place 🤔 since we can't change the search attribute since there's only one